apiary.apib

```diff
@@ -248,11 +248,75 @@ You can find a (very) minimal PHP code example, showing how start the OAuth 2 au
 
 A Teamleader PHP SDK is under development. In the meanwhile, we recommend using packages such [Guzzle](http://docs.guzzlephp.org/en/stable) or the [League OAuth2 client](http://oauth2-client.thephpleague.com).
 
-## Changelog
+## Changes & upgrades
 
-You can follow our API progress and ongoing proposals via our [API Blueprint GitHub repository](https://github.com/teamleadercrm/api).
+We want to keep this API backwards compatible as long as possible, but at the same time we need to make changes to it to match our evolving product.
+
+Therefor we assigned a dated version to your API client when it was created.
+This version controls the API and webhooks behaviour for you API client.
+When we change the API in a backwards-incompatible way, we release a new dated version.
+But to avoid breaking your code, you stay on your assigned version until you [upgrade your API version](./#upgrading-your-api-version).
+
+### Types of changes
+
+We consider the following changes to be backwards-compatible:
+
+- Adding new API endpoints.
+- Adding new optional parameters to existing API requests.
+- Adding new properties to existing API responses.
+- Adding new webhook types.
+- Making existing parameters optional.
+- Changing the order of properties in existing API responses.
+- Fixing (potential) bugs that made an endpoint unusable (HTTP 500 errors).
+- (Currently we also hold ourselves the right to return a different error format, as there is still room for improvement)
```

Comment:
PP: comma after currently

Reply:
Also: do we really want to do this without introducing another API version? I've seen @mrtus having to emergency fix an integration because another service did that and I don't think we want to be the people doing the same thing.

Reply:
But making error handling backwards compatible, seems too much effort for value.
Let's advise people not to hardcode on error contents and we can still see...

Reply:
😅 Sometimes you just need to have proper error handling to react on exceptions mostly based on error contents. 

If you need information on why exactly feel free to pass by, happy to discuss.

Reply:
gonna keep it in here, we can still decide to remove it...